### PR TITLE
implement rest metrics

### LINKED ISSUE #1768: API Ratelimit Event and Request Counter
Hello, dear D#+ team,

I would really like to see an event when the rate limit is hit, just as some stats about how close I am to the ratelimit.

Even though some may think that this is nonsense, it is a nice-to-have feature. It may also help with diagnosis.

In detail I request:

1. An Event in the DiscordClient class which is triggered when the Ratelimit is (almost) hit.
2. Some APIState Class that will return the current allowed ratelimit, and how close I am to it. (In %)

I hope this feature may be added.
Thanks.

### PR DESCRIPTION
**Pull request by akiraveliara**
resolves #1768 

Implements rest metrics, as per the documentation article, on all standalone clients. This supports both lifetime metrics and periodic tracking.

### CODE REVIEW

**Comment by Plerx2493** [comment]
Currently you only expose it as a string, is there a reason to not expose it as a readonly copy? Im thinking about something like a systems which checks for certain prozentages automaticly without string parsing

**Review by InFTord — COMMENTED** [review]
from documentation side, lgtm

File: DSharpPlus/Metrics/LiveRestCollection.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
+namespace DSharpPlus.Metrics;
+
+internal record struct LiveRestCollection
```
Comment (Plerx2493) [inline DSharpPlus/Metrics/LiveRestCollection.cs:3]: maybe rename to ´LiveRestMetricCollection´ so you dont have to look at the namespace when seeing in the wild. Also i would use either RequestMetrics **or** RestMetrics.

**Review by Plerx2493 — APPROVED** [review]
Looks good to me, just one naming nit

**Pull request merged**